### ISSUE TITLE
Single column, empty row CSV files result in empty rows and truncation of the last row

### ISSUE THREAD

**Issue by rkennedy-mode** [uniVocity/univocity-parsers#336]
We're parsing some CSV files with Univocity where the files have a single column and all of the rows are empty. When this happens, we notice two things:

1. The `String[]` returned by `parseNext()` is zero length. The null value that we should be getting isn't present.
2. If we have `N` empty rows in the CSV file, we only get back `N - 1` rows from the parser.

I've created a reproduction case against `com.univocity:univocity-parsers:2.8.1`:

```java
import com.univocity.parsers.csv.CsvParser;
import com.univocity.parsers.csv.CsvParserSettings;

import java.io.ByteArrayInputStream;
import java.io.InputStreamReader;
import java.nio.charset.Charset;

class UnivocityWrongCountOnEmptyLines {
    public static void main(String[] args) {
        final String inputData = "\"HEADER_NAME\"\n" +
                "\n" + // row  1
                "\n" + // row  2
                "\n" + // row  3
                "\n" + // row  4
                "\n" + // row  5
                "\n" + // row  6
                "\n" + // row  7
                "\n" + // row  8
                "\n" + // row  9
                "";    // row 10

        final CsvParserSettings settings = new CsvParserSettings();
        settings.setHeaderExtractionEnabled(true);
        settings.setMaxCharsPerColumn(-1);
        settings.setMaxColumns(1_000);
        settings.setSkipEmptyLines(false);

        final CsvParser inputParser = new CsvParser(settings);
        inputParser.beginParsing(new InputStreamReader(new ByteArrayInputStream(inputData.getBytes(Charset.forName("utf-8")))));

        System.out.println(String.format("Found %d column(s): %s",
                inputParser.getRecordMetadata().headers().length,
                String.join(", ", inputParser.getRecordMetadata().headers())));

        int rowCount = 0;
        String[] row;
        while ((row = inputParser.parseNext()) != null) {
            rowCount++;
            System.out.println(String.format("Row %d; %d columns; values = %s",
                    rowCount, row.length, String.join(", ", row)));
        }
        System.out.println("rowCount = " + rowCount);
    }
}
```

When the above is run the results will be as follows:

```
Found 1 column(s): HEADER_NAME
Row 1; 0 columns; values = 
Row 2; 0 columns; values = 
Row 3; 0 columns; values = 
Row 4; 0 columns; values = 
Row 5; 0 columns; values = 
Row 6; 0 columns; values = 
Row 7; 0 columns; values = 
Row 8; 0 columns; values = 
Row 9; 0 columns; values = 
rowCount = 9
```

If I change the `inputData` to include a single comma per line, things change dramatically as the parser seems to start doing the right thing:

```java
        final String inputData = "\"HEADER_NAME_ONE\",\"HEADER_NAME_TWO\"\n" +
                ",\n" + // row  1
                ",\n" + // row  2
                ",\n" + // row  3
                ",\n" + // row  4
                ",\n" + // row  5
                ",\n" + // row  6
                ",\n" + // row  7
                ",\n" + // row  8
                ",\n" + // row  9
                ",";    // row 10
```

The output now shows the `null` column values and doesn't clip the 10th row:

```
Found 2 column(s): HEADER_NAME_ONE, HEADER_NAME_TWO
Row 1; 2 columns; values = null, null
Row 2; 2 columns; values = null, null
Row 3; 2 columns; values = null, null
Row 4; 2 columns; values = null, null
Row 5; 2 columns; values = null, null
Row 6; 2 columns; values = null, null
Row 7; 2 columns; values = null, null
Row 8; 2 columns; values = null, null
Row 9; 2 columns; values = null, null
Row 10; 2 columns; values = null, null
rowCount = 10
```

**Comment by rkennedy-mode** [comment]
The clipping of the 10th row can most likely be explained as not being part of the dataset since the row wasn't terminated by the end of line delimiter.

**Comment by jbax** [comment]
Thank you for reporting the bug. If running with `setSkipEmptyLines(false);`, you should get rows of length 1 with the null value in them. I've fixed the error and released a 2.8.3-SNAPSHOT build that includes the adjustment.

As you mentioned, that there is no 10th row being clipped on your first input as the operation `+ "";` does not append anything to the string. Here the parser is producing the correct number of rows.

Thank you for using our parsers
